File: generate_tree.js
const fs = require('fs');
const path = require('path');

const excludeDirs = ['node_modules', '.git'];

function getStructure(dir, prefix = '') {
    let result = '';
    try {
        const items = fs.readdirSync(dir);

        // Sort items: directories first, then files
        items.sort((a, b) => {
            const aPath = path.join(dir, a);
            const bPath = path.join(dir, b);
            const aStat = fs.statSync(aPath);
            const bStat = fs.statSync(bPath);

            if (aStat.isDirectory() && !bStat.isDirectory()) return -1;
            if (!aStat.isDirectory() && bStat.isDirectory()) return 1;
            return a.localeCompare(b);
        });

        for (let i = 0; i < items.length; i++) {
            const item = items[i];
            if (excludeDirs.includes(item)) continue;

            const itemPath = path.join(dir, item);
            const stats = fs.statSync(itemPath);
            const isLast = i === items.length - 1;

            // Use simpler indentation for clarity or standard tree style
            // result += prefix + (isLast ? '└── ' : '├── ') + item + '\n';
            // For deep nesting, simpler indentation might be better copy-paste
            result += prefix + item + (stats.isDirectory() ? '/' : '') + '\n';

            if (stats.isDirectory()) {
                // Recurse
                // result += getStructure(itemPath, prefix + (isLast ? '    ' : '│   '));
                result += getStructure(itemPath, prefix + '    ');
            }
        }
    } catch (e) {
        result += prefix + `[Error reading directory: ${e.message}]\n`;
    }
    return result;
}

// Find the project directory
const cwd = process.cwd();
const files = fs.readdirSync(cwd);
let targetDir = cwd;

// If we are in the root and 'nextjs14-aitoonic-done...' exists, go in there
for (const f of files) {
    if (f.startsWith('nextjs14-aitoonic-done') && fs.statSync(f).isDirectory()) {
        targetDir = path.join(cwd, f);
        break;
    }
}

console.log(`Generating key for: ${targetDir}`);
const tree = getStructure(targetDir);
fs.writeFileSync('full_structure.txt', tree);
console.log('Tree written to full_structure.txt');

File: generate_tree.py
import os

def generate_tree(startpath, exclude_dirs=None):
    if exclude_dirs is None:
        exclude_dirs = []
    
    tree_str = []
    
    for root, dirs, files in os.walk(startpath):
        # Modify dirs in-place to skip excluded directories
        dirs[:] = [d for d in dirs if d not in exclude_dirs]
        
        level = root.replace(startpath, '').count(os.sep)
        indent = ' ' * 4 * (level)
        tree_str.append('{}{}/'.format(indent, os.path.basename(root)))
        subindent = ' ' * 4 * (level + 1)
        for f in files:
            tree_str.append('{}{}'.format(subindent, f))
            
    return '\n'.join(tree_str)

if __name__ == "__main__":
    # Exclude node_modules and .git as they are too large and not relevant for project structure understanding
    # .next is included as requested (implied by "start from .next") but beware it might be large
    excludes = ['node_modules', '.git'] 
    
    # Current directory
    cwd = os.getcwd()
    # Go into the main project folder if we are one level up, or stay here
    # Based on previous ls, the code is in 'nextjs14-aitoonic-done-...'
    # We should search dynamically or assuming we run this from relevant root.
    
    target_dir = '.'
    # Check if we are in the parent and need to go into the project folder
    items = os.listdir('.')
    for item in items:
        if item.startswith('nextjs14-aitoonic-done') and os.path.isdir(item):
            target_dir = item
            break
            
    print(f"Generating tree for: {os.path.abspath(target_dir)}")
    print(generate_tree(target_dir, excludes))
